src/platform/ESP32/OTAImageProcessorImpl.cpp

```diff
@@ -280,6 +280,36 @@ esp_err_t OTAImageProcessorImpl::DeltaOTAWriteCallback(const uint8_t * buf, size
 }
 #endif // CONFIG_ENABLE_DELTA_OTA
 
+#if defined(CONFIG_AUTO_UPDATE_RCP) && defined(CONFIG_OPENTHREAD_BORDER_ROUTER)
+esp_err_t OTAImageProcessorImpl::ProcessRcpImage(const uint8_t * buffer, uint32_t bufLen)
+{
+    esp_err_t err = ESP_OK;
+
+    if (!this->mRcpFirmwareDownloaded)
```

Comment:
```suggestion
    if (!mRcpFirmwareDownloaded)
```
And others in this function.